src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +31,48 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        String[] addressComponents = trimmedAddress.split(",");
+
+
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        processAddress(addressComponents);
+
+    }
+
+    /**
+     * Initialise all address components according to input
+     *
+     * @param addressComponents
+     */
+
+    private void processAddress(String[] addressComponents){
+
+        this.numberOfAddressComponents = addressComponents.length;
+
+        for(int i = 0; i < this.numberOfAddressComponents; i++){
+            switch(i){
```

Comment:
Indentation problem here? Check the Java coding standards for switch-case indentation.